polaris-react/src/styles/shared/_responsive-props.scss

```diff
@@ -1,10 +1,28 @@
+@mixin scope-custom-property(
+  $componentName,
+  $componentProp,
+  $responsively: false
+) {
+  @if $responsively {
+    // stylelint-disable -- Polaris component custom properties
+    --pc-#{$componentName}-#{$componentProp}-xs: initial;
+    --pc-#{$componentName}-#{$componentProp}-sm: initial;
+    --pc-#{$componentName}-#{$componentProp}-md: initial;
+    --pc-#{$componentName}-#{$componentProp}-lg: initial;
+    --pc-#{$componentName}-#{$componentProp}-xl: initial;
+  } @else {
+    --pc-#{$componentName}-#{$componentProp}: initial;
+  }
+}
+
+@mixin non-responsive-prop($componentName, $componentProp, $declarationProp) {
+  @include scope-custom-property($componentName, $componentProp);
+
+  #{$declarationProp}: var(--pc-#{$componentName}-#{$componentProp});
+}
```

Comment:
My take here is to remove this mixin and include `@include scope-custom-property` where you need it. In the future we'll have atomic css from our tokens and shouldn't need all these custom properties everywhere. This added mixin complicates that clean up work.

The previous css was also much easier to read

Reply:
Agreed this is obfuscating some things, but it's also enforcing a basic level of standardization.

My main goal was to follow the established pattern with `@mixin responsive-props()`, the motivation being to avoid this issue coming back up again in the future (I can imagine even myself seeing `--pc-foo-whatever: initial` and thinking "That's completely useless, let's be a good engineer and remove it" without realizing it's important. Similarly for naming of breakpoint props - I actually _want_ that encapsulated away somewhere (mixin or function, I don't mind) so it can be consistent with little effort from the dev (again, to avoid future accidental errors being introduced).

Maybe I'm being overly cautious, but the fact this bug existed for so long without anyone noticing implies to me if we accidentally reintroduce it somewhere, it'll go unnoticed again for a while.

> In the future we'll have atomic css from our tokens

Does that involve switching to a CSS-in-JS solution? Because without it, we'll still need to do this "donut scope" encapsulation of the CSS variables step regardless of what they're named.

Reply:
> Does that involve switching to a CSS-in-JS solution

Yes and no, I guess. I'm thinking we need css classes created from tokens, so we'd apply a class (based off of a prop) for padding for example and would automatically be scoped. This would clean up a ton of our css and remove a lot of buggy custom properties